src/App.tsx

```diff
@@ -2,10 +2,46 @@ import React from 'react'
 import logo from './logo.svg'
 import './App.css'
 import FileImportButton from './Components/FileImportButton'
+import TextField from '@material-ui/core/TextField'
+import xml2js from 'xml2js'
+import Attempt from './types'
 
 const App = () => {
-  function handleFile() {
-    console.log('test')
+  const [attemptHistory, setAttemptHistory] = React.useState<Attempt[]>([])
+
+  function handleFile(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e?.target?.files?.[0]) {
+      const file = e.target.files[0]
+
+      const reader = new FileReader()
+      reader.onload = () => {
+        const xml = reader.result?.toString()
+
+        if (xml) {
+          xml2js.parseString(xml, { trim: true }, async function(err, result) {
+            //todo: display error nicely in UI
+            if (err) {
+              console.log(err)
+            }
+
+            await getAttemptHistory(result)
+          })
+        }
+      }
+      reader.readAsText(file)
+    }
+  }
+
+  async function getAttemptHistory(xml: any) {
```

Comment:
This kinda of wrapper function is tricky to name, but it's a bit funny to be to have a method called `getAttemptHistory` that internally calls `setAttemptHistory` in this way (is this a getter or a setter?).

I might extract the formatting portion of this method into a method like `parseAttemptHistoryFile` which returns the `Attempt[]` value which you then directly call `setAttemptHistory()` on _here_ to remove that extra layer of removal that getAttemptHistory currently injects.

Reply:
Also, technically this method _doesn't_ take in `xml`, it takes in the parsed file results so this name is a bit conflated with the `xml` variable above.

Reply:
updated this function name to be parseAttemptHistoryFromFile, and pass the result of calling it to setAttemptHistory